core/risk/engine.go

```diff
@@ -142,9 +151,50 @@ func NewEngine(log *logging.Logger,
 		stateVarEngine.NewEvent(asset, mktID, statevar.EventTypeMarketEnactment)
 	}
 
+	e.updateMarginLevels = e.bufferMarginLevels
+	if e.StreamMarginLevelsVerbose {
+		e.updateMarginLevels = e.sendMarginLevels
+	}
+
 	return e
 }
 
+func (e *Engine) FlushMarginLevelsEvents() {
+	if e.StreamMarginLevelsVerbose || len(e.marginLevelsUpdates) <= 0 {
```

Comment:
This is a mega nit. The length can't be less then zero. So I think `len(e.marginLevelsUpdates) < 1` would do.